gitlab/v4/objects/epics.py

```diff
@@ -29,6 +30,63 @@ class GroupEpic(ObjectDeleteMixin, SaveMixin, RESTObject):
     resourcelabelevents: GroupEpicResourceLabelEventManager
     notes: GroupEpicNoteManager
 
+    def _epic_path(self) -> str:
+        """Return the API path for this epic using its real group."""
+        if not hasattr(self, "group_id") or self.group_id is None:
+            raise AttributeError(
+                "Cannot compute epic path: attribute 'group_id' is missing."
+            )
+        encoded_group_id = gitlab.utils.EncodedId(self.group_id)
+        return f"/groups/{encoded_group_id}/epics/{self.encoded_id}"
+
+    @exc.on_http_error(exc.GitlabUpdateError)
+    def save(self, **kwargs: Any) -> dict[str, Any] | None:
+        """Save the changes made to the object to the server.
+
+        The object is updated to match what the server returns.
+
+        This method uses the epic's group_id attribute to construct the correct
+        API path. This is important when the epic was retrieved from a parent
+        group but actually belongs to a sub-group.
+
+        Args:
+            **kwargs: Extra options to send to the server (e.g. sudo)
+
+        Returns:
+            The new object data (*not* a RESTObject)
+
+        Raises:
+            GitlabAuthenticationError: If authentication is not correct
+            GitlabUpdateError: If the server cannot perform the request
+        """
+        # Use the epic's actual group_id to construct the correct path.
+        path = self._epic_path()
+
+        # Call SaveMixin.save() method
+        return super().save(_pg_custom_path=path, **kwargs)
```

Comment:
Not sure if we should have some more generic `_get_custom_path` or `_resolve_path` helpers that would automatically get picked up by the mixins, as we already do this twice for delete/update. That way we don't need an override here (or maybe also in some other places), but I'm not sure if feasible or worth it.